nette/notejam/app/presenters/NotePresenter.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace App\Presenters;
+
+use App\Forms\Note\DeleteNoteFormFactory;
+use App\Forms\Note\EditNoteFormFactory;
+use App\Forms\Note\NewNoteFormFactory;
+use App\Model\NoteManager;
+use App\Model\PadManager;
+use Nette;
+use App\Model;
+use Nette\Application\BadRequestException;
+
+
+class NotePresenter extends SecuredBasePresenter
+{
+
+	/** @var PadManager @inject */
+	public $padManager;
+
+	/** @var NoteManager @inject */
+	public $noteManager;
+
+	/** @var EditNoteFormFactory @inject */
+	public $editPadFormFactory;
+
+	/** @var DeleteNoteFormFactory @inject */
+	public $deletePadFormFactory;
+
+	/** @var NewNoteFormFactory @inject */
+	public $newPadFormFactory;
+
+	/** @var int */
+	private $id;
+
+	/** @var int */
+	private $padId;
+
+	/** @var object */
+	private $note;
+
+	/** @var object[] */
+	private $notes;
+
+
+	/********************* Controls ***********************************************************************************/
+
+	/**
+	 * @return \Nette\Application\UI\Form
+	 */
+	protected function createComponentEditNoteForm()
+	{
+		$form = $this->editPadFormFactory->create($this->id, $this->note->name, $this->note->text, $this->note->pad_id);
+		$form->onError[] = function ($form) {
+			foreach ($form->getErrors() as $error) {
+				$this->flashMessage($error, 'error');
+			}
+		};
+		$form->onSuccess[] = function ($form) {
+			$this->flashMessage('Note successfully updated', 'success');
+			$form->getPresenter()->redirect('default', ['id' => $this->id]);
+		};
+		return $form;
+	}
+
+	/**
+	 * @return \Nette\Application\UI\Form
+	 */
+	protected function createComponentDeleteNoteForm()
+	{
+		$form = $this->deletePadFormFactory->create($this->id);
+		$form->onError[] = function ($form) {
```

Comment:
If it is general message about an event it's fine to use negative flash message (like "Invalid username or password"). If it is field specific error flash messages is not the best choice.